File: languages/python/fibonacci.py
import sys

_cache = {}
def cache(func):
    def wrap(n):
        if n not in _cache:
            _cache[n] = func(n)
        return _cache[n]
    return wrap

@cache
def fib(n): 
    return n if n < 2 else fib(n-1) + fib(n-2)

if __name__ == '__main__':
    print fib(int(sys.argv[1]) - 1)

File: languages/c/fibonacci.c
#include <stdio.h>

int main(int argc, char *argv[]){
	printf("%d\n", fib(atoi(argv[1])));
	return 0;
}

int fib(int n){
	return n < 2 ? n : fib(n-1) + fib(n-2);
}